### PR TITLE
Task/client filter

### PR DESCRIPTION
**Pull request by JuanGro**
#### What's this PR do?
To integrate the search in clients component.

#### Where should the reviewer start?
With the way to find an item.

#### How should this be manually tested?
Searching an specific contact by name and address.

#### Any background context you want to provide?
No

This template was adapted ~stolen~ from [Quickleft/Sprint.ly](https://quickleft.com/blog/pull-request-templates-make-code-review-easier/)

### CODE REVIEW

File: src/app/accounts/clients/clients.component.ts [hunk 1]
```diff
@@ -22,8 +22,16 @@ import { environment } from '../../../environments/environment';
...
+  @Input('currentClient') currentClient: Client;
+  // Received from table component, it gives me the filtered client list if the user is using the search.
+  @Input('clientListFiltered') clientListFiltered: Client[];
+  // Received from table component, it gives me the value that the user is typing in the search.
+  @Input('valueSearch') valueSearch: string;
   // Variable that saves the title to show in the template.
   public title: string;
+  // Original copy of the client list received from httpService.
+  public completeClientList: Client[];
   // Client list
```
Comment (EduardoVaca) [inline src/app/accounts/clients/clients.component.ts]: Explain better this variable, its important

File: src/app/accounts/clients/clients.component.ts [hunk 2]
```diff
@@ -85,59 +95,100 @@ export class ClientsComponent implements OnInit {
...
+    this.clientsList = [];
+    this.completeClientList.sort();
+    if (this.valueSearch === '') {
+      this.clientsList = this.completeClientList;
+    } else {
+      for (let clientFromList of this.completeClientList) {
+        let client = new Client(clientFromList);
+        if (client.name.toLowerCase().includes(this.valueSearch.toLowerCase()) ||
+            client.address.toLowerCase().includes(this.valueSearch.toLowerCase())) {
+            this.clientsList.push(client);
```
Comment (EduardoVaca) [inline src/app/accounts/clients/clients.component.ts]: Create the new client just if it is going to be pushed to the completeList.
Example: this.client.psuh(new Client(clientFromList))

File: src/app/accounts/clients/clients.component.ts [hunk 3]
```diff
@@ -85,59 +95,100 @@ export class ClientsComponent implements OnInit {
...
+    this.valueSearch = value;
+    this.clientsList = [];
+    this.completeClientList.sort();
+    if (this.valueSearch === '') {
+      this.clientsList = this.completeClientList;
+    } else {
+      for (let clientFromList of this.completeClientList) {
+        let client = new Client(clientFromList);
+        if (client.name.toLowerCase().includes(this.valueSearch.toLowerCase()) ||
+            client.address.toLowerCase().includes(this.valueSearch.toLowerCase())) {
```
Comment (EduardoVaca) [inline src/app/accounts/clients/clients.component.ts]: Remove filter using the client ADDRESS

File: src/app/accounts/clients/clients-list.component.ts [hunk 4]
```diff
@@ -51,4 +53,12 @@ export class ClientsListComponent {
   public sendCurrentClient(object: Client) {
     this.currentClient.emit(object);
   }
+
+  /**
+  * Search specific items in the contacts list,
```
Comment (EduardoVaca) [inline src/app/accounts/clients/clients-list.component.ts]: clients, not contacts

File: src/app/accounts/clients/clients.component.ts (hunk 3 above)
Comment (JuanGro) [inline src/app/accounts/clients/clients.component.ts]: Done

File: src/app/accounts/clients/clients.component.ts (hunk 2 above)
Comment (JuanGro) [inline src/app/accounts/clients/clients.component.ts]: Done

File: src/app/accounts/clients/clients.component.ts (hunk 1 above)
Comment (JuanGro) [inline src/app/accounts/clients/clients.component.ts]: Done

File: src/app/accounts/clients/clients-list.component.ts (hunk 4 above)
Comment (JuanGro) [inline src/app/accounts/clients/clients-list.component.ts]: Done

File: src/app/accounts/clients/clients.component.ts [hunk 5]
```diff
@@ -22,8 +22,16 @@ import { environment } from '../../../environments/environment';
 * - Remove a client.
 **/
 export class ClientsComponent implements OnInit {
+  // Received from table component, it gives me the contact that the user selected to see his detail.
+  @Input('currentClient') currentClient: Client;
+  // Received from table component, it gives me the filtered client list if the user is using the search.
+  @Input('clientListFiltered') clientListFiltered: Client[];
```
Comment (EduardoVaca) [inline src/app/accounts/clients/clients.component.ts]: Are you removing this?

File: src/app/accounts/clients/clients.component.ts [hunk 6]
```diff
@@ -22,8 +22,16 @@ import { environment } from '../../../environments/environment';
...
+  // Received from table component, it gives me the value that the user is typing in the search.
+  @Input('valueSearch') valueSearch: string;
   // Variable that saves the title to show in the template.
   public title: string;
+  // Original copy of the client list it's used always like a base for filter by name.
+  public completeClientList: Client[];
   // Client list
```
Comment (EduardoVaca) [inline src/app/accounts/clients/clients.component.ts]: Explain better

**Pull request merged**